engine/model_draft_transactions.go

```diff
@@ -343,39 +291,131 @@ func (m *DraftTransaction) createTransactionHex(ctx context.Context) (err error)
 		}
 
 		m.Configuration.Outputs[0].Scripts[0].Satoshis = m.Configuration.Outputs[0].Satoshis
-	} else {
-		if satoshisReserved < satoshisNeeded+fee {
-			return ErrNotEnoughUtxos
+		return nil
+	}
+
+	if satoshisReserved < satoshisNeeded+fee {
+		return ErrNotEnoughUtxos
+	}
+
+	// if we have a remainder, add that to an output to our own wallet address
+	satoshisChange := satoshisReserved - satoshisNeeded - fee
+	m.Configuration.Fee = fee
+	if satoshisChange > 0 {
+		var newFee uint64
+		newFee, err := m.setChangeDestination(
+			ctx, satoshisChange, fee,
+		)
+		if err != nil {
+			return err
 		}
+		m.Configuration.Fee = newFee
+	}
 
-		// if we have a remainder, add that to an output to our own wallet address
-		satoshisChange := satoshisReserved - satoshisNeeded - fee
-		m.Configuration.Fee = fee
-		if satoshisChange > 0 {
-			var newFee uint64
-			newFee, err = m.setChangeDestination(
-				ctx, satoshisChange, fee,
-			)
-			if err != nil {
-				return
-			}
-			m.Configuration.Fee = newFee
+	return nil
+}
+
+func (m *DraftTransaction) prepareUtxos(ctx context.Context, opts []ModelOps, satoshisNeeded uint64) (*[]*bt.UTXO, uint64, error) {
+	if m.Configuration.SendAllTo != nil { // Send TO ALL
+		inputUtxos, satoshisReserved, err := m.prepareSendAllToUtxos(ctx, opts)
+		if err != nil {
+			return nil, 0, err
 		}
+		return inputUtxos, satoshisReserved, nil
 	}
 
-	// Add the outputs to the bt transaction
-	if err = m.addOutputsToTx(tx); err != nil {
-		return
+	inputUtxos, satoshisReserved, err := m.prepareSeparateUtxos(ctx, opts, satoshisNeeded)
```

Comment:
In this PR, I guess, we have to test it carefully, because changes involve making transaction.
So I suppose we should test:
- making transaction from spv-wallet user to another spv-wallet user
- making transaction from spv-wallet to external wallet
- making OP_RETURN transaction

Check if transactions have been broadcasted and mined
Check if the balance is ok